sdv/multi_table/base.py

```diff
@@ -161,3 +163,59 @@ def validate(self, data):
 
         if errors:
             raise InvalidDataError(errors)
+
+    def _fit(self, processed_data):
+        """Fit the model to the tables.
+
+        Args:
+            processed_data (dict):
+                Dictionary mapping each table name to a preprocessed ``pandas.DataFrame``.
+        """
+        raise NotImplementedError()
+
+    def preprocess(self, data):
+        """Transform the raw data to numerical space.
+
+        Args:
+            data (dict):
+                Dictionary mapping each table name to a ``pandas.DataFrame``.
+
+        Returns:
+            dict:
+                A dictionary with the preprocessed data.
+        """
+        self.validate(data)
+        if self._fitted:
+            warnings.warn(
+                'This model has already been fitted. To use the new preprocessed data, '
+                "please refit the model using 'fit' or 'fit_processed_data'."
```

Comment:
I think right now if someone runs `fit` twice in a row, they will see this warning even though they shouldn't